### ISSUE TITLE
Test discovery failing from 0.3.5

### ISSUE THREAD

**Issue by pchomik** [kondratyev-nv/vscode-python-test-adapter#156]
Hello,

I back to your adapter and I discovered that plugin is not working for me. Last working version is 0.3.4.

**Environment**
```
Package                       Version
----------------------------- -----------
atomicwrites                  1.4.0
attrs                         19.3.0
backports.functools-lru-cache 1.6.1
configparser                  4.0.2
contextlib2                   0.6.0.post1
funcsigs                      1.0.2
importlib-metadata            1.6.1
more-itertools                5.0.0
packaging                     20.4
pathlib2                      2.3.5
pip                           20.1
pluggy                        0.13.1
py                            1.8.1
pyparsing                     2.4.7
pytest                        4.6.11
pytest-describe               1.0.0
scandir                       1.10.0
setuptools                    44.1.0
six                           1.15.0
wcwidth                       0.2.4
wheel                         0.34.2
zipp                          1.2.0
```

**PyEnv**
* pyenv 1.2.18

**Test 1**
```
def test_false():
    assert True is False
```

**Test 2**
```
def describe_test_fail():
    def it_test_fail():
        assert True is False
```

**Output from working version 0.3.4**
```
2020-06-08T05:41:25.143Z info at 'test' [unittest runner]: Reading configuration for workspace test
2020-06-08T05:41:25.148Z info at 'test' [unittest runner]: Unittest test discovery is disabled
2020-06-08T05:41:25.148Z info at 'test' [pytest runner]: Reading configuration for workspace test
2020-06-08T05:41:25.153Z info at 'test' [pytest runner]: Environment variables file /home/chomicki/Repos/github/test/.env does not exist
2020-06-08T05:41:25.154Z info at 'test' [pytest runner]: Discovering tests using python path 'python' in /home/chomicki/Repos/github/test
```

**Output from not working latest version**
```
2020-06-08T05:42:24.103Z info at 'test' [unittest runner]: Reading configuration for workspace test
2020-06-08T05:42:24.105Z info at 'test' [unittest runner]: Unittest test discovery is disabled
2020-06-08T05:42:24.105Z info at 'test' [pytest runner]: Reading configuration for workspace test
2020-06-08T05:42:24.217Z info at 'test' [pytest runner]: Environment variables file /home/test/Repos/github/test/.env does not exist
2020-06-08T05:42:24.218Z info at 'test' [pytest runner]: Discovering tests using python path '/home/test/.pyenv/versions/testvscode/bin/python' in test
2020-06-08T05:42:24.218Z info at 'test' [pytest runner]: Running pytest wrapper with arguments: --collect-only
2020-06-08T05:42:24.237Z crit at 'test' [pytest runner]: Test loading failed: Error occurred during process execution: Error: spawn /home/test/.pyenv/versions/testvscode/bin/python ENOENT
```

**Python settings in vscode**
```
{
    "python.pythonPath": "/home/test/.pyenv/versions/testvscode/bin/python"
}
```

I checked the script embedded in the plugin and it is working if executed from separate file.
I added some logging to plugin but looks like script is not executed at all because I don't see any debug logs (I wrote them into file).
I'm on Manjaro latest version and I'm using **zsh** shell.

Thank you in advance for support.
Pawel

**Comment by Hanspagh** [comment]
I have the same problem

**Comment by kondratyev-nv** [comment]
@pchomik Thank you for the issue! From the logs it looks like you're using different Python executables (just `python` in the first log and `/home/test/.pyenv/versions/testvscode/bin/python` in the second). Can you try to use `python` executable from virtualenv or from your system? Just to check if it works, maybe there are some incompatibility with pyenv. I will also try to reproduce the problem with pyenv.

> I checked the script embedded in the plugin and it is working if executed from separate file.
I added some logging to plugin but looks like script is not executed at all because I don't see any debug logs (I wrote them into file).

Are you talking about pytest-describe plugin? 

@Hanspagh Do you have the same problem? Do you use pyenv as well?

**Comment by Hanspagh** [comment]
@kondratyev-nv I am using poetry, it had been working for me for a while, but as of today, it stopped working. I can see it is not exactly the same error
```
2020-06-08T14:34:18.597Z info at 'caf' [unittest runner]: Reading configuration for workspace caf
2020-06-08T14:34:18.598Z info at 'caf' [unittest runner]: Unittest test discovery is disabled
2020-06-08T14:34:18.599Z info at 'caf' [pytest runner]: Reading configuration for workspace caf
2020-06-08T14:34:18.600Z info at 'caf' [pytest runner]: Environment variables file /Users/hanspagh/Documents/Code/wondercast/caf/.env does not exist
2020-06-08T14:34:18.600Z info at 'caf' [pytest runner]: Discovering tests using python path '/Users/hanspagh/Library/Caches/pypoetry/virtualenvs/caf-TOl5zFmu-py3.7' in /Users/hanspagh/Documents/Code/wondercast/caf
2020-06-08T14:34:18.601Z info at 'caf' [pytest runner]: Running pytest wrapper with arguments: --collect-only,-vv
2020-06-08T14:34:18.607Z crit at 'caf' [pytest runner]: Test loading failed: Error occurred during process execution: Error: spawn /Users/hanspagh/Library/Caches/pypoetry/virtualenvs/caf-TOl5zFmu-py3.7 EACCES
```
I get the EACCES error instead, the test discovery is working for regular python tets explorer

**Comment by kondratyev-nv** [comment]
@Hanspagh I'm new to poetry, but following their example and setting `python.pythonPath` to `/Users/<username>/Library/Caches/pypoetry/virtualenvs/poetry-demo-U7oW9SQj-py2.7/bin/python` worked for me just fine. As I can see in your log messages your `python.pythonPath` is set to `/Users/hanspagh/Library/Caches/pypoetry/virtualenvs/caf-TOl5zFmu-py3.7` which in my case is a folder. Python executable should be a little bit further in `/Users/hanspagh/Library/Caches/pypoetry/virtualenvs/caf-TOl5zFmu-py3.7/bin/python` and `python.pythonPath` expects a path to an executable. (However, I'm not sure setting `python.pythonPath` to some hidden path to a cache is a good idea).

Please, try to set `python.pythonPath` to a path to a Python executable.

**Comment by Hanspagh** [comment]
I will try. Everything else in vs code works with that path though, so it
is a bit odd

**Comment by kondratyev-nv** [comment]
@Hanspagh Yeah, I've looked through the Python extension documentation and it looks like `python.pythonPath` can be set to a folder. I'll try to support it as well, but it looks like this issue does not have a lot in common with the original one, so I've created another - https://github.com/kondratyev-nv/vscode-python-test-adapter/issues/157.

**Comment by Hanspagh** [comment]
Thank you 😊

**Comment by pchomik** [comment]
Hello,

I changed my setup to pure virtualenv. I created virtualenv in the project in `venv` directory. Now my settings looks as follow:

**.vscode/settings.json**
```
{
    "python.pythonPath": "/home/chomicki/Repos/github/test/venv/bin/python"
}
```

**output from0.3.4**
```
2020-06-09T05:34:19.433Z info at 'test' [unittest runner]: Reading configuration for workspace test
2020-06-09T05:34:19.435Z info at 'test' [unittest runner]: Unittest test discovery is disabled
2020-06-09T05:34:19.436Z info at 'test' [pytest runner]: Reading configuration for workspace test
2020-06-09T05:34:19.442Z info at 'test' [pytest runner]: Environment variables file /home/user/Repos/github/test/.env does not exist
2020-06-09T05:34:19.443Z info at 'test' [pytest runner]: Discovering tests using python path '/home/user/Repos/github/test/venv/bin/python' in /home/chomicki/Repos/github/test
```

**output from 0.4.2**
```
2020-06-09T05:35:59.049Z info at 'test' [unittest runner]: Reading configuration for workspace test
2020-06-09T05:35:59.050Z info at 'test' [unittest runner]: Unittest test discovery is disabled
2020-06-09T05:35:59.051Z info at 'test' [pytest runner]: Reading configuration for workspace test
2020-06-09T05:35:59.156Z info at 'test' [pytest runner]: Environment variables file /home/user/Repos/github/test/.env does not exist
2020-06-09T05:35:59.156Z info at 'test' [pytest runner]: Discovering tests using python path '/home/user/Repos/github/test/venv/bin/python' in test
2020-06-09T05:35:59.157Z info at 'test' [pytest runner]: Running pytest wrapper with arguments: --collect-only
2020-06-09T05:35:59.180Z crit at 'test' [pytest runner]: Test loading failed: Error occurred during process execution: Error: spawn /home/user/Repos/github/test/venv/bin/python ENOENT
```

I have same result. Version  0.3.4 is working. Version 0.4.2 not.

This issue is not about pytest-describe. I have to tests: one with test function, one with describe functions. I put both examples to have full view where is the issue.

**Comment by kondratyev-nv** [comment]
@pchomik Well, I've checked how the extension works with virtualenv, and don't see any issues. Usually, ENOENT error means that an executable not found. Does file `/home/user/Repos/github/test/venv/bin/python` exist? I've also tried to install version 0.3.4 and test discovery works fine (execution, however, fails). Do you have any other values in settings? Like `python.testing.pytestArgs`, for example.

**Comment by pchomik** [comment]
The path is correct because if working manually and is working with 0.3.4 version.

I have a lot of values set globally. I will re-test with fresh vscode configuration.

**Comment by pchomik** [comment]
With fresh copy of vscode the latest version is working. I need to review my configuration and find wrong configuration.

I double checked with virtualenv and **pyenv** and the plugin is working.

Thank you for support.

**Comment by kondratyev-nv** [comment]
@pchomik Glad that you've got it working! I think I've got an idea of what is going wrong. One of the parts that differ in logs is the current working directory.
`Discovering tests using python path '...' in test`
vs
`Discovering tests using python path '...' in /home/user/Repos/github/test`
I've tried to set `python.testing.cwd` to a non-existing path and got the same error - ENOENT. So changing `python.testing.cwd` might help. And just to make sure, can you describe your folder structure? What folder is opened in VSCode and does `test` folder exists under the folder you've opened?

**Comment by pchomik** [comment]
@kondratyev-nv the configuration `python.testing.cwd` was a problem. I found that I set this value as global and for workspace. Everything is working after removing this value.

In general all my packages have structure as follow:
```
package_name (root)
    package_name (as source)
    test (contains tests)
    setup.py
```

I think I set value to `test` to eliminate entries from source code (because linters are enabled).
Is it possible to support which python.testing.cwd path as relative to the project?
